server/tests/integration_tests/test_get_nearby_users.py

```diff
@@ -25,3 +25,53 @@ def test_get_potential_neighbors_no_results(user, user2, user3, user4, login, cl
     
     response_json = json.loads(response.data)
     assert(len(response_json) == 0)
+
+
+def test_add_neighbor(user, user2, login, client):
+    """Should return user 2."""
+    client.put(
+        f'user/add_neighbor',
+        json=dict(current_user_id=user.unique_identity, neighbor_id=user2.unique_identity),
+        content_type="application/json"
+    )
+
+    # Potential neighbors should be empty
+    response = client.get(
+        f'user/nearby/{user.unique_identity}'
+    )
+    response_json = json.loads(response.data)
+    assert(len(response_json) == 0)
+
+
+def test_remove_neighbor(user, user2, login, client):
+    """Should not return results (user 4 is also a provider and should not be returned)."""
+    client.put(
+        f'user/add_neighbor',
+        json=dict(current_user_id=user.unique_identity, neighbor_id=user2.unique_identity),
+        content_type="application/json"
+    )
+
+    client.put(
+        f'user/remove_neighbor',
+        json=dict(current_user_id=user.unique_identity, neighbor_id=user2.unique_identity),
+        content_type="application/json"
+    )
+
+    # Neighbors should be empty
+    response = client.get(
+        f'user/neighbors/{user.unique_identity}'
+    )
+    
+    response_json = json.loads(response.data)
+    assert(len(response_json) == 0)
+
+    # Potential neighbors should be empty
+    response = client.get(
+        f'user/nearby/{user.unique_identity}'
+    )
+    response_json = json.loads(response.data)
+    assert(len(response_json) == 0)
+
+    user_model = User.objects.get(unique_identity=user.unique_identity)
+    assert(len(user_model.blacklisted_neighbors) == 1)
+
```

Comment:
Depending on what we decide id vs unique_identity, we might need to update these tests.